src/library/blast/congruence_closure.cpp

```diff
@@ -280,7 +280,8 @@ static optional<ext_congr_lemma> to_ext_congr_lemma(name const & R, expr const &
 }
 
 static optional<ext_congr_lemma> mk_ext_congr_lemma_core(name const & R, expr const & fn, unsigned nargs) {
-    simp_rule_set const * sr = get_simp_rule_sets(env()).find(R);
+    simp_rule_sets srss = mk_simp_rule_sets(env());
```

Comment:
There was a trade-off. It was more convenient to implement it on the arithmetic branch, since I was implementing it at the time just to be able to use `div` lemmas as simp rules for `inv`, and the tests I was using depending on various things in the arithmetic branch. The issue is that the arithmetic branch was still very much in flux, and once it stabilized it would have required some git-fu to separate out the simp rule part. It did not strike me as essential to do this at the time, but I suppose I made the wrong decision. Since you implemented the simp rule normalization yourself on master, I'll remove this part of the commit.